src/java.base/share/classes/java/lang/Process.java

```diff
@@ -149,12 +169,85 @@ public abstract class Process {
+     * invoking this method has no effect.
+     * <p>
+     * Before calling this method the caller should read the streams for any
+     * data or text and call {@linkplain #waitFor() waitFor} if the exit value is needed.
+     * The contents of streams that have not been read fully are lost,
+     * they are discarded or ignored.
+     * Streams should be {@code closed} when no longer needed.
+     * Closing an already closed stream usually has no effect but is specific to the stream.
+     * If an {@code IOException} occurs when closing a stream it is
+     * re-thrown after the process is terminated. Additional {@code IOExceptions}
+     * thrown by closing the remaining streams, if any, are added to the first
+     * {@code IOException} as {@linkplain IOException#addSuppressed suppressed exceptions}.
+     * <p>
+     * The process may already have terminated or be in the process of terminating;
+     * if it is {@linkplain #isAlive() alive}, it is {@linkplain #destroy destroyed}.
+     * On some platforms, {@linkplain #supportsNormalTermination() normal termination}
+     * is not available and the process is forcibly terminated.
+     * Calling {@link #waitFor() waitFor} before calling {@code close} or exiting
+     * the try-with-resources block allows the process time to clean up and exit.
+     * <p>
+     * Try-with-resources example to write text to a process, read back the
+     * response, and close the streams and process:
+     * {@snippet class=ProcessExamples region=example}
+     *
+     * @implSpec
+     * The {@code outputWriter} and {@code outputStream} to the process are closed.
+     * The {@code inputReader} and {@code inputStream} from the process are closed.
+     * The {@code errorReader} and {@code errorStream} from the process are closed.
+     * The process is terminated.
+     * @throws IOException if closing any of the streams throws an exception
+     * @since 26
+     */
+    @Override
+    public void close() throws IOException {
+        synchronized(this) {
+            if (closed) {
+                return;
+            }
+            closed = true;
+            // Close each stream
+            IOException ioe = quietClose(outputWriter != null ? outputWriter : getOutputStream(), null);
+            ioe = quietClose(inputReader != null ? inputReader : getInputStream(), ioe);
+            ioe = quietClose(errorReader != null ? errorReader : getErrorStream(), ioe);
+
+            destroy();      // no-op if process is not alive
+            if (ioe != null) {
+                throw ioe;
+            }
+        }
+    }
+
+    // Quietly close.
+    // If an IOException occurs and it is the first, return it.
+    // Otherwise, add the exception as a suppressed exception to the first.
+    private IOException quietClose(Closeable c, IOException firstIOE) {
+        try {
+            c.close();
+            return firstIOE;
+        } catch (IOException ioe) {
+            if (firstIOE == null) {
```

Comment:
Maybe update to:
```suggestion
            if (firstIOE == null || firstIOE == ioe) {
```
To avoid edge case failure?